broker/src/main/java/org/apache/rocketmq/broker/client/ProducerManager.java

```diff
@@ -185,4 +191,33 @@ public void unregisterProducer(final String group, final ClientChannelInfo clien
             log.error("", e);
         }
     }
+
+    public Channel getAvaliableChannel(String groupId) {
+        HashMap<Channel, ClientChannelInfo> channelClientChannelInfoHashMap = groupChannelTable.get(groupId);
+        List<Channel> channelList = new ArrayList<Channel>();
+        if (channelClientChannelInfoHashMap != null) {
+            for (Channel channel : channelClientChannelInfoHashMap.keySet()) {
+                channelList.add(channel);
+            }
+            int size = channelList.size();
+            if (0 == size) {
+                log.warn("Channel list is empty. groupId={}", groupId);
+                return null;
+            }
+
+            int index = positiveAtomicCounter.incrementAndGet() % size;
+            Channel channel = channelList.get(index);
+            int count = 0;
+            boolean isOk = channel.isActive() && channel.isWritable();
+            while (isOk && count++ < GET_AVALIABLE_CHANNEL_RETRY_COUNT) {
+                index = (++index) % size;
+                channel = channelList.get(index);
+                return channel;
+            }
```

Comment:
here if isOk= true   will return next channel.
i think  it should be like 
`while ( count++ < GET_AVALIABLE_CHANNEL_RETRY_COUNT) {
               if(isOk){
                      return channel;
                }
                index = (++index) % size;
               channel = channelList.get(index);
               isOk = channel.isActive() && channel.isWritable();
            }`

Reply:
Thanks for your careful review，and I will fix this problem ASAP.